spec/integration/apply_spec.rb

```diff
@@ -103,4 +173,98 @@ def root_config
       end
     end
   end
+
+  describe 'over winrm', winrm: true do
+    let(:uri) { conn_uri('winrm') }
+    let(:password) { conn_info('winrm')[:password] }
+    let(:user) { conn_info('winrm')[:user] }
+
+    def config
+      { 'modulepath' => File.join(__dir__, '../fixtures/apply'),
+        'winrm' => {
+          'ssl' => false,
+          'ssl-verify' => false,
+          'user' => conn_info('winrm')[:user],
+          'password' => conn_info('winrm')[:password]
+        } }
+    end
+
+    context "when running against puppet 5" do
```

Comment:
Are you relying on the puppet_agent::install task doing an upgrade here?

Reply:
Also, with Windows tests. I think we can probably leave out the acceptance tests. These integration tests should be enough to catch incompatibilities in Puppet.

Reply:
From manual testing on vmpooler instances when installing from an MSI or DMG the install task can handle moving back and forth between puppet 5 and 6 collections. It is only the linux package managers that needed special handling for the upgrade (and will fail to downgrade) across collections.